fifteen_min/src/find_amenities.rs

```diff
@@ -0,0 +1,143 @@
+use map_gui::tools::{ChooseSomething, ColorLegend};
+use map_gui::ID;
+use map_model::AmenityType;
+use widgetry::{
+    Cached, Choice, Color, Drawable, EventCtx, GfxCtx, HorizontalAlignment, Key, Line, Panel,
+    SimpleState, State, Transition, VerticalAlignment, Widget,
+};
+
+use crate::isochrone::{Isochrone, Options};
+use crate::viewer::{draw_star, draw_unwalkable_roads, HoverKey, HoverOnBuilding};
+use crate::App;
+
+/// Calculate isochrones around each amenity on a map and merge them together using the min value
+pub struct FindAmenity {
+    options: Options,
+}
+
+impl FindAmenity {
+    pub fn new(ctx: &mut EventCtx, options: Options) -> Box<dyn State<App>> {
+        ChooseSomething::new(
+            ctx,
+            "Choose an amenity",
+            AmenityType::all()
+                .into_iter()
+                .map(|at| Choice::new(at.to_string(), at))
+                .collect(),
+            Box::new(move |at, ctx, app| {
+                let multi_isochrone = create_multi_isochrone(ctx, app, at, options.clone());
+                return Transition::Push(Results::new(ctx, app, multi_isochrone, at));
+            }),
+        )
+    }
+}
+
+/// For every one of the requested amenity on the map, draw an isochrone
+fn create_multi_isochrone(
+    ctx: &mut EventCtx,
+    app: &App,
+    category: AmenityType,
+    options: Options,
+) -> Isochrone {
+    let map = &app.map;
+    // For a category, find all matching stores
+    let mut stores = Vec::new();
+    for b in map.all_buildings() {
+        if b.has_amenity(category) {
+            stores.push(b.id);
+        }
+    }
+    Isochrone::new(ctx, app, stores, options.clone())
+}
+
+struct Results {
+    draw: Drawable,
+    isochrone: Isochrone,
+    hovering_on_bldg: Cached<HoverKey, HoverOnBuilding>,
+    // TODO Can't use Cached due to a double borrow
+    hovering_on_category: Option<(AmenityType, Drawable)>,
+    draw_unwalkable_roads: Drawable,
+}
+
+impl Results {
+    fn new(
+        ctx: &mut EventCtx,
+        app: &App,
+        isochrone: Isochrone,
+        category: AmenityType,
+    ) -> Box<dyn State<App>> {
+        let panel = Panel::new(Widget::col(vec![
+            Widget::row(vec![
+                Line(format!("{} within 15 minutes", category))
+                    .small_heading()
+                    .into_widget(ctx),
+                ctx.style().btn_close_widget(ctx),
+            ]),
+            ColorLegend::categories(
```

Comment:
I think it'd be useful to just print the number of matching stores here. I found myself trying to count the number of stars on the map. You could do something like `format!("{} matching stores", isochrone.start.len()).text_widget(ctx)`. That's a shortcut to go directly from a `String` to rendering text. (The longer form is `Text::from(vec![Line(format!(...)]).into_widget(ctx)`

Reply:
Yes I added it. text_widget is handy!